### ISSUE TITLE
torch.cuda.set_rng_state() followed by torch.cuda.get_rng_state() return different result

### ISSUE THREAD

**Issue by hrbigelow** [pytorch/pytorch#20843]
## 🐛 Bug

```python
torch.cuda.set_rng_state(stored_state)
retrieved_state = torch.cuda.get_rng_state() 
```

stored_state and retrieved_state differ in 400 positions out of 824016 in:
pytorch 1.0.0
Cuda 10.0.130
GeForce GTX 1070 Max-Q

they **are** the same in:
pytorch 1.1.0
Cuda 10.0.130
Tesla K80

## To Reproduce

\# contents of rng_test.sh
```python
python - <<EOF
import torch
torch.save(torch.cuda.get_rng_state(), 'cuda_rng_state.tch')
torch.save(torch.get_rng_state(), 'rng_state.tch')
EOF

python - <<EOF
import torch
cuda_pre_state = torch.load('cuda_rng_state.tch')
pre_state = torch.load('rng_state.tch')
torch.cuda.set_rng_state(cuda_pre_state)
torch.set_rng_state(pre_state)
cuda_cur_state = torch.cuda.get_rng_state()
cur_state = torch.get_rng_state()
print('cuda state tensors differ in {} positions out of {}.'.format(
    cuda_pre_state.ne(cuda_cur_state).sum(),
    cuda_pre_state.nelement()))
print('cpu state tensors differ in {} positions out of {}.'.format(
    pre_state.ne(cur_state).sum(),
    pre_state.nelement()))

EOF
```
$ . rng_test.sh 
cuda state tensors differ in **400** positions out of **824016**.
cpu state tensors differ in **0** positions out of **5048**.

## Expected behavior

I would expect the states to be the same.

## Environment
This is the environment failing the test:
PyTorch version: 1.0.0
Is debug build: No
CUDA used to build PyTorch: 10.0.130

OS: Ubuntu 18.04.2 LTS
GCC version: (Ubuntu 6.5.0-2ubuntu1~18.04) 6.5.0 20181026
CMake version: version 3.10.2

Python version: 3.6
Is CUDA available: Yes
CUDA runtime version: Could not collect
GPU models and configuration: GPU 0: GeForce GTX 1070 with Max-Q Design
Nvidia driver version: 410.104
cuDNN version: /usr/lib/x86_64-linux-gnu/libcudnn.so.7.4.2

Versions of relevant libraries:
[pip] numpy==1.15.4
[pip] torch==1.0.0
[pip] torchvision==0.2.1
[conda] blas                      1.0                         mkl  
[conda] cuda100                   1.0                           0    pytorch
[conda] mkl                       2018.0.3                      1  
[conda] mkl_fft                   1.0.6            py36h7dd41cf_0  
[conda] mkl_random                1.0.1            py36h4414c95_1  
[conda] pytorch                   1.0.0           py3.6_cuda10.0.130_cudnn7.4.1_1  [cuda100]  pytorch
[conda] torchvision               0.2.1                      py_2    pytorch
(pytorch) henry@henry-gs65:~$ 

## Additional context

In this environment, the test works properly.
PyTorch version: 1.1.0
Is debug build: No
CUDA used to build PyTorch: 10.0.130

OS: Debian GNU/Linux 9.9 (stretch)
GCC version: (Debian 6.3.0-18+deb9u1) 6.3.0 20170516
CMake version: Could not collect

Python version: 3.7
Is CUDA available: Yes
CUDA runtime version: 10.0.130
GPU models and configuration: GPU 0: Tesla K80
Nvidia driver version: 410.72
cuDNN version: Could not collect

Versions of relevant libraries:
[pip3] intel-numpy==1.15.1
[pip3] numpy==1.15.1
[pip3] torch==1.1.0
[pip3] torchvision==0.2.2.post3
[conda] blas                      1.0                         mkl  
[conda] mkl                       2019.3                      199  
[conda] mkl-service               1.1.2            py37he904b0f_5  
[conda] mkl_fft                   1.0.10           py37ha843d7b_0  
[conda] mkl_random                1.0.2            py37hd81dba3_0  
[conda] pytorch                   1.1.0           py3.7_cuda10.0.130_cudnn7.5.1_0    pytorch
[conda] torchvision               0.2.2                      py_3    pytorch

**Comment by soumith** [comment]
since this is already fixed in the latest release (1.1.0) I am closing the issue.
We are unlikely to port the fix to 1.0.0

**Comment by hrbigelow** [comment]
No problem, thanks for checking it.  I'll upgrade to 1.1.0.
